packages/tools/devtools/devtools-browser-extension/src/background/BackgroundScript.ts

```diff
@@ -43,6 +43,26 @@ type Port = chrome.runtime.Port;
 
 console.log(formatBackgroundScriptMessageForLogging("Initializing Background Worker."));
 
+// Establish communication channels with the PopupScript and ContentScript.
+let contentScriptPort: Port;
```

Comment:
Note that there is a single background script instance associated with **all** tabs, popups, etc. If we want to associate the port with a specific tab ID (which I think we will need to differentiate tab ports correctly), you'll need to manage a map of some kind here. You should be able to subscribe to some sort of "on disconnect" event to know when to remove tabId/port entries.